SonarQube.TeamBuild.Integration/TeamBuildSettings.cs

```diff
@@ -72,7 +74,8 @@ public static TeamBuildSettings GetSettingsFromEnvironment(ILogger logger)
                         BuildEnvironment = env,
                         BuildUri = Environment.GetEnvironmentVariable(EnvironmentVariables.BuildUri_Legacy),
                         TfsUri = Environment.GetEnvironmentVariable(EnvironmentVariables.TfsCollectionUri_Legacy),
-                        BuildDirectory = Directory.GetCurrentDirectory()
+                        BuildDirectory = Environment.GetEnvironmentVariable(EnvironmentVariables.BuildDirectory_Legacy),
+                        AnalysisBaseDirectory = Directory.GetCurrentDirectory()
```

Comment:
The property that was called "BuildDirectory" is now called "AnalysisBaseDirectory". I haven't changed how this is calculated as part of this fix; it still uses the current directory.

"BuildDirectory" returns the build directory set by the build system, or null if not running under TeamBuild. This is checked by the unit tests.